orchagent/tunneldecaporch.cpp

```diff
@@ -803,3 +940,35 @@ IpAddresses TunnelDecapOrch::getDstIpAddresses(std::string tunnelKey)
 
     return tunnelTable[tunnelKey].dst_ip_addrs;
 }
+
+/**
+ * Function Description:
+ *    @brief Resolve the map id from QosOrch
+ *
+ * Arguments:
+ *    @param[in] tunnle_name - The name of tunnel
+ *    @param[in] map_type_name - The type of referenced QoS map
+ *    @param[in] tuple - The KeyOpFieldsValuesTuple that contains keys - values
+ *
+ * Return Values:
+ *    @return The sai_object_id of referenced map, or SAI_NULL_OBJECT_ID  if there's an error
+ */
+sai_object_id_t TunnelDecapOrch::resolveQosMapId(std::string tunnle_name, std::string map_type_name, KeyOpFieldsValuesTuple& tuple)
+{
+    sai_object_id_t id;
+    string object_name;
+    ref_resolve_status status = resolveFieldRefValue(QosOrch::getTypeMap(), map_type_name, qos_to_ref_table_map.at(map_type_name), tuple, id, object_name);
```

Comment:
As mentioned, it is not a good idea to use qos data structure in another orch

Reply:
Fixed.